docs/preview/02-Features/assertion.md

```diff
@@ -0,0 +1,210 @@
+
+### Customization
+The test assertion also expose several options to tweak the behavior of the XML comparison.
+
+```csharp
+using Arcus.Testing;
+
+AssertXml.Equal(..., options =>
+{
+    // Adds one ore more local names of XML nodes that should be excluded from the XML comparison.
+    options.IgnoreNode("local-node-name");
+
+    // Sets the type of order which should be used when comparing XML attributes.
+    // REMARK: only the order of XML attributes can be set, XML elements are still compared by their contents.
+    // Default: Ignore.
+    options.Order = AssertXmlOrder.Include;
+
+    // Sets the maximum characters of the expected and actual inputs should be written to the test output.
+    // Default: 500 characters.
+    options.MaxInputCharacters = 1000;
+});
+```
+
+### Loading XML documents yourself
+The XML assertion equalization can be called directly with raw contents - internally it parses to a valid XML structure: `XmlDocument`. It it so happens that you want to compare two XML nodes with different serialization settings, you can load the two nodes separately and do the equalization on the loaded nodes.
+
+💡 It provides you with more options to control how your file should be loaded.
+
+```csharp
+using System.Xml;
+using Arcus.Testing;
+
+string xml = ...;
+
+XmlDocument expected = AssertXml.Load(json);
+
+// Or with options (same as available with `AssertJson.Equal`).
+XmlDocument actual = AssertXml.Load(json, options => ...);
+
+// Use overload with nodes instead.
+AssertXml.Equal(expected, actual);
+```
+
+> ❓ Why use the `AssertXml` to load something that is already available with `XmlDocument.LoadXml`?
+
+The `AssertXml.Load` is a special variant on the existing load functionality in such a way that it provides more descriptive information on the input file that was trying to be parsed, plus by throwing an assertion message, it makes the test output more clear on what the problem was and where it happened. 
+
+## JSON
+The library has an `AssertJson` class that exposes useful test assertions when dealing with JSON outputs. The most popular one is comparing JSON documents.
+
+```csharp
+using Arcus.Testing;
+
+string expected = "{ \"root\": ... }";
+string actual = "{ \"diff-root\": ... }";
+
+AssertJson.Equal(expected, actual);
+// Arcus.Testing.EqualAssertionException
+// AssertJson.Equal failure: expected and actual JSON contents do not match
+// Actual JSON misses property at $.root
+//
+// Expected:
+// {
+//    "root": ...
+// }
+//
+// Actual:
+// {
+//    "diff-root": ...
+// }
+```
+
+💡 Currently, the input contents are trimmed in case if the input is too big to be shown in a humanly readable manner to the test output. In case of large files, it might be best to log those files (or parts that interest you) separately before using this test assertion.
```

Comment:
Maybe remove the "if" from the first sentence to simplify a bit:
"Currently, the input contents are trimmed _in case the input_"


Also I was wondering, do we provide examples on how to loge the files separately (what's explained in the second sentence)? If yes maybe showcase or explain how this can be done?

Reply:
1. Yes, that will more more readable.
2. No, no examples as of yet; this is also very test suite-specific as the type of logging or the place that the test output is written could differ. In Azure DevOps, you could for example publish it as an artifact, maybe; but again, it is very project-specific. The warning here is more like to point out that large files might need a different approach. Food for thought, I believe.

Reply:
Fixed 1. here: https://github.com/arcus-azure/arcus.testing/pull/110/commits/2f4a53836ac20ef7a298b88517df30ed7cf653e2